cadwyn/schema_generation.py

```diff
@@ -151,16 +152,12 @@ def generate_field_copy(self, generator: "SchemaGenerator") -> pydantic.fields.F
         )
 
 
-def _extract_passed_field_attributes(field_info: FieldInfo):
-    attributes = {
-        attr_name: field_info._attributes_set[attr_name]
-        for attr_name in _all_field_arg_names
-        if attr_name in field_info._attributes_set
+def _extract_passed_field_attributes(field_info: FieldInfo) -> dict[str, object]:
```

Comment:
@Viicos WDYT about this approach for extracting metadata? Is this the correct interface to use?

Reply:
If I understand correctly this is meant to "reconstruct" the kwargs passed to `FieldInfo`? It is still relying on private logic (`_attributes_set`, which I'm hoping to remove at some point, and `collect_known_metadata()`, which isn't guaranteed to give metadata that is available as `FieldInfo` kwargs).

As it is, I wouldn't say this approach is robust. Maybe can find a better solution to your problem but I think I will need more context as to why this logic is necessary in the first place.

Reply:
I see, I think there are two ways to approach this:
- Using the private constructs described in my previous comment. Pros: in sync with every Pydantic version if we ever add new attributes to the field function. Cons: can break at any point.
- Defining the existing attributes explicitly (e.g. keeping a list of strings representing the attributes, and also a list of constraints in the `metadata` list that should be converted back to `FieldInfo` kwargs). Pros: not relying on private logic. Cons: this known attributes/metadata list will need to be updated on the cadwyn side.